File: flutter_app/lib/features/users/data/models/update_user_status_request.dart
/// Request body for PATCH /users/{id}/status.
///
/// Mirrors the backend request field: account_status ("Active" | "Inactive").
class UpdateUserStatusRequest {
  const UpdateUserStatusRequest({required this.accountStatus});

  final String accountStatus;

  Map<String, dynamic> toJson() => {'account_status': accountStatus};
}

File: backend/app/Http/Requests/Users/UpdateUserStatusRequest.php
<?php

namespace App\Http\Requests\Users;

use Illuminate\Foundation\Http\FormRequest;

class UpdateUserStatusRequest extends FormRequest
{
    public function authorize(): bool
    {
        return true;
    }

    public function rules(): array
    {
        return [
            'account_status' => ['required', 'string', 'in:Active,Inactive'],
        ];
    }

    public function messages(): array
    {
        return [
            'account_status.required' => 'Account status is required.',
            'account_status.in' => 'The selected account_status is invalid.',
        ];
    }
}
